### PR TITLE
Add compatibility with PHPUnit 9.3

### PR DESCRIPTION
**Pull request by villfa**
This is a required step to build the project with PHP8.

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Will conflict with https://github.com/guzzle/promises/pull/113.

**Comment by villfa** [comment]
@GrahamCampbell I have reworked the PR hoping this will make the merge easier

**Comment by Nyholm** [comment]
Could you please rebase this PR?

**Comment by Nyholm** [comment]
I started working on PHP8 another ago. I see that that PR is using PHPUnit 9.3.10 with the symfony phpunit bridge. 

Have a look at #122. What are the pros/cons between that approach?

File: tests/FulfilledPromiseTest.php [hunk 1]
```diff
@@ -1,11 +1,10 @@
 <?php
 
-namespace GuzzleHttp\Tests\Promise;
+namespace GuzzleHttp\Promise\Tests;
```
Comment (Nyholm) [inline tests/FulfilledPromiseTest.php:3]: Good catch!

File: tests/TestCase.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace GuzzleHttp\Promise\Tests;
+
+use PHPUnit\Framework\TestCase as PHPUnitTestCase;
+
+class TestCase extends PHPUnitTestCase
+{
+    public function setExpectedException($exception, $message = null, $code = null)
```
Comment (Nyholm) [inline tests/TestCase.php:9]: This is actually the job of Symfony PHPUnit bridge.

File: tests/UtilsTest.php [hunk 3]
```diff
@@ -403,7 +397,7 @@ public function testCanCatchAndYieldOtherException()
         $promise->otherwise(function ($value) use (&$result) { $result = $value; });
         P\Utils::queue()->run();
         $this->assertTrue(P\Is::rejected($promise));
-        $this->assertContains('foo', $result->getMessage());
+        $this->assertTrue(strpos($result->getMessage(), 'foo') !== false, "'" . $result->getMessage() . " does not contain 'foo'");
```
Comment (Nyholm) [inline tests/UtilsTest.php:400]: assertStringContainsString

**Review by Nyholm — CHANGES_REQUESTED** [review]

**Comment by villfa** [comment]
> I started working on PHP8 another ago. I see that that PR is using PHPUnit 9.3.10 with the symfony phpunit bridge.
> 
> Have a look at #122. What are the pros/cons between that approach?

When I have originally open this PR I wanted to keep it as minimal as possible.
Now that *phpunit-bridge* has been added I think it's not relevant to maintain this PR anymore.
I'm closing it in favor of yours.

**Comment by Nyholm** [comment]
Yeah, I appreciate that. 

This morning when I started with this library I had no plans of creating PHP8 support. 

Thank you for you work.

**Pull request closed without merging**